src/mapgen.cpp

```diff
@@ -717,16 +717,28 @@ class jmapgen_toilet : public jmapgen_piece {
 class jmapgen_gaspump : public jmapgen_piece {
 public:
     jmapgen_int amount;
+    std::string fuel;
     jmapgen_gaspump( JsonObject &jsi ) : jmapgen_piece()
-    , amount( jsi, "amount", 0, 0 )
+    , amount( jsi, "amount", 0, 0 ), fuel("")
     {
+        if (jsi.has_string("fuel")){
+            fuel = jsi.get_string("fuel");
+
+            // may want to not force this, if we want to support other fuels for some reason
+            if (fuel != "gasoline" && fuel != "diesel"){
+                fuel = "gasoline";
```

Comment:
I think it would be better to throw an error here instead:
```C++
jsi.throw_error("invalid fuel", "fuel");
```

You never know what the author had in mind, maybe they wrote "disel" and meant "diesel".